webplugin/js/app/media/media-service.js

```diff
@@ -41,39 +81,54 @@ Kommunicate.mediaService = {
                     )
                 );
             };
-            recognition.onerror = function (err) {
-                console.log('error while speech recognition', err);
-            };
             recognition.onend = function () {
                 // stop mic effect
+                recognition.stop();
                 Kommunicate.typingAreaService.hideMiceRecordingAnimation();
                 window.$applozic.fn.applozic('toggleMediaOptions');
             };
+            recognition.onerror = function (event) {
+                console.log('error while speech recognition', event.error);
+                recognition.abort();
+            };
+
+            //explicitly Stop the Mic recording only for Safari
+            if (isAppleProduct) {
+                var checkLastSpeech = setInterval(function () {
```

Comment:
will this keep running even if user is not using voice?

Reply:
no, 
1> It only runs whenever the user has trying to access the widget speech-to-text function from an Apple device
2>  once it enables, it continues to run every 1 sec until the condition matches then we cleared this interval hence set interval stops

Reply:
Uploading Screen Recording 2023-04-23 at 1.08.17 PM.mov…